hr_attendance_calendar/models/hr_attendance_day.py

```diff
@@ -0,0 +1,370 @@
+##############################################################################
+#
+#    Copyright (C) 2018 Compassion CH (http://www.compassion.ch)
+#    @author: Eicher Stephane <seicher@compassion.ch>
+#
+#    The licence is in the file __manifest__.py
+#
+##############################################################################
+
+from datetime import date, timedelta
+
+from odoo import models, fields, api
+
+
+class HrAttendanceDay(models.Model):
+    """
+    The instances of hr.attendance.day is created either at the first
+    attendance of the day or by the method
+    hr.employee._cron_create_attendance() called by a cron everyday.
+    """
+    _name = "hr.attendance.day"
+    _order = 'date DESC'
+    _sql_constraints = [('unique_product', 'unique(date, employee_id)',
+                         'This "Attendance day" already exist for this '
+                         'employee')]
+
+    ##########################################################################
+    #                                 FIELDS                                 #
+    ##########################################################################
+    date = fields.Date("Date",
+                       # readonly=True, todo: uncomment
+                       required=True)
+    employee_id = fields.Many2one('hr.employee', "Employee",
+                                  ondelete="cascade", required=True)
+
+    # Working schedule
+    cal_att_ids = fields.Many2many('resource.calendar.attendance',
+                                   string="Working schedule",
+                                   required=True)
+    working_day = fields.Char("Working day", compute='_compute_working_day',
+                              readonly=True, store=True)
+    name = fields.Char(compute='_compute_name', store=True)
+
+    # Leaves
+    leave_ids = fields.Many2many('hr.holidays', readonly=True, string='Leaves')
+    in_leave = fields.Boolean('In leave', compute='_compute_in_leave',
+                              store=True)
+    public_holiday_id = fields.Many2one('hr.holidays.public.line',
+                                        'Public holidays')
+
+    # Due hours
+    due_hours = fields.Float('Due hours', compute='_compute_due_hours',
+                             readonly=True, store=True)
+
+    # Attendances
+    attendance_ids = fields.One2many('hr.attendance', 'attendance_day_id',
+                                     'Attendances', readonly=True)
+
+    # Worked
+    worked_hours = fields.Float('Worked hours',
+                                compute='_compute_worked_hours', store=True,
+                                readonly=True)
+    coefficient = fields.Float(readonly=True, string='Coefficient',
+                               help='Worked hours coefficient')
+
+    # Break
+    due_break_min = fields.Float('Minimum break due',
+                                 compute='_compute_due_break')
+    due_break_total = fields.Float('Total break due',
```

Comment:
I would give another name as `total`. Here you use that for the long break in case the employee worked more than 9 hours if I'm correct? Or why not having only one field named `due_break` that would take the long break or the small break depending on the worked hours? This would be more easy to understand in my opinion.